pkg/registry/core/service/strategy.go

```diff
@@ -390,6 +405,16 @@ func dropTypeDependentFields(newSvc *api.Service, oldSvc *api.Service) {
 		}
 	}
 
+	// If a user is switching to a type that doesn't need LoadBalancerClass AND they did not change
+	// this field, it is safe to drop it.
+	if oldSvc.Spec.Type == api.ServiceTypeLoadBalancer && newSvc.Spec.Type != api.ServiceTypeLoadBalancer {
```

Comment:
Please write helpers like needsNodePort above (yes, the allocatedLoadBalancerNodePorts block is ugly, don't copy it)